components/team_controller/index.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect, ConnectedProps} from 'react-redux';
+import {RouteComponentProps} from 'react-router-dom';
+
+import {fetchAllMyTeamsChannelsAndChannelMembersREST, fetchMyChannelsAndMembersREST, viewChannel} from 'mattermost-redux/actions/channels';
+import {getLicense, getConfig} from 'mattermost-redux/selectors/entities/general';
+import {getCurrentTeamId, getMyTeams} from 'mattermost-redux/selectors/entities/teams';
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+import {getCurrentChannelId} from 'mattermost-redux/selectors/entities/channels';
+import {isGraphQLEnabled} from 'mattermost-redux/selectors/entities/preferences';
+
+import {GlobalState} from 'types/store';
+
+import {getSelectedThreadIdInCurrentTeam} from 'selectors/views/threads';
+
+import {markChannelAsReadOnFocus} from 'actions/views/channel';
+import {initializeTeam, joinTeam} from 'components/team_controller/actions';
+import {fetchChannelsAndMembers} from 'actions/channel_actions';
+
+import {checkIfMFARequired} from 'utils/route';
+
+import TeamController from './team_controller';
+
+type Params = {
+    url: string;
+    team?: string;
+}
+
+export type OwnProps = RouteComponentProps<Params>;
+
+function mapStateToProps(state: GlobalState, ownProps: OwnProps) {
+    const license = getLicense(state);
+    const config = getConfig(state);
+    const currentUser = getCurrentUser(state);
+    const plugins = state.plugins.components.NeedsTeamComponent;
+    const graphQLEnabled = isGraphQLEnabled(state);
+
+    return {
+        currentUser,
+        currentTeamId: getCurrentTeamId(state),
+        currentChannelId: getCurrentChannelId(state),
+        teamsList: getMyTeams(state),
+        plugins,
+        selectedThreadId: getSelectedThreadIdInCurrentTeam(state),
+        mfaRequired: checkIfMFARequired(currentUser, license, config, ownProps.match.url),
+        graphQLEnabled,
+    };
+}
+
+const mapDispatchToProps = {
+    fetchChannelsAndMembers,
+    fetchMyChannelsAndMembersREST,
+    fetchAllMyTeamsChannelsAndChannelMembersREST,
+    viewChannel,
+    markChannelAsReadOnFocus,
+    initializeTeam,
+    joinTeam,
+};
```

Comment:
I saw in a different file you using `useSelector` instead of this. 0/5 on what is better / more desirable.

Reply:
this is just a way. For any new file we are following the useSelector. for older ones keeping as they were